src/keboola/facebook/extractor/query.clj

```diff
@@ -25,7 +25,9 @@
 
 (defn retrieve-page-access-token [id token version]
   (let [accounts (swallow-exceptions (request/get-accounts token :version version))
-        page-token (if (some? accounts) (:access_token (first (filter #(= id (:id %)) accounts))))]
+        page-token-from-accounts-list (if (some? accounts) (:access_token (first (filter #(= id (:id %)) accounts))))
+        page-token-from-page-details (swallow-exceptions (request/get-page-token-via-page-details token id))
+        page-token (or page-token-from-accounts-list page-token-from-page-details)]
     (if (nil? page-token)
       (runtime/log-strings "Could not find page access token for" id ". Token from the configuration will be used instead.")
       (runtime/log-strings "Using page access token to retrieve data for" id))
```

Comment:
I know it's not part of the change, but isn't the second log message here a lie?  the first message says that it couldn't find the page access token, and the second one says that it's using the page access token (that it couldn't find?

Reply:
good question, actually this is `if` syntax of clojure, first expression in the list is if the condition passes, the second expression in the list is if the condition fails (then expression) so it is like if-then-else but the else keyword is missing here :)